### PR TITLE
changeable attributes names + auto provisioning on-off option

### PR DESCRIPTION
**Pull request by restena-ma**
- add configuration options for to choose SAML attributes name (defaults on previous)
- permit to bypass auto-provisioning of unknown accounts (user should exists on plainauth db)
- compatibility with  SimpleSAMLPhp tested ok

### CODE REVIEW

**Comment by splitbrain** [comment]
Can you explain the rationale behind this PR? Why would you want to use different attributes?

File: conf/metadata.php [hunk 1]
```diff
@@ -1,8 +1,13 @@
...
+*/
 
 $meta['endpoint']    = array('string');
 $meta['certificate'] = array('');
 $meta['lowercase']   = array('onoff');
+$meta['autoprovisioning'] = array('onoff');
+$meta['userid_attr_name'] = array('');
+$meta['fullname_attr_name'] = array('');
+$meta['email_attr_name'] = array('');
+$meta['groups_attr_name'] = array('');
```
Comment (splitbrain) [inline conf/metadata.php]: these should all be of type string

**Comment by restena-ma** [comment]
Attributes can by named differently depending on the idP conventions.

In the case of our organization, the attributes are named in respect of the OASIS recommendations:
[OASIS SAML2 attributes naming](https://docs.oasis-open.org/security/saml/SpecDrafts-Post2.0/sstc-saml-attribute-x500-cd-01.html) cf. "2.3 SAML Attribute Naming"

Unlocking the attributes name permits to use this extension with many other SAML2 idP such as SimpleSAMLphp, OneLogin, PingFederate which are all widely used.

**Comment by splitbrain** [comment]
Makes sense. Thanks for the PR. Would you mind documenting your config at the plugin's page?

**Pull request merged**